### PR TITLE
feat: suppport config.cluster.https

### PR DESCRIPTION
**Pull request by atian25**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

##### Description of change

https://github.com/eggjs/egg/issues/1870

### CODE REVIEW

**Comment by dead-horse** [comment]
单测？

**Comment by atian25** [comment]
https 那个现在没单测的，@suyi 补下？

**Comment by thonatos** [comment]
@atian25 @dead-horse 

本地测试完了，晚点补上去。

File: test/https.test.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+'use strict';
```
Comment (atian25) [inline test/https.test.js]: 好像之前有一个 skip 的单测，可以顺便干掉

**Comment by fengmk2** [comment]
rebase master

**Comment by thonatos** [comment]
done.

**Comment by atian25** [comment]
@dead-horse Review 下

**Pull request merged**